dpdispatcher/shell.py

```diff
@@ -23,11 +23,17 @@ def do_submit(self, job):
         job_id_name = job.job_hash + '_job_id'
         self.context.write_file(fname=script_file_name, write_str=script_str)
         proc = self.context.call('cd %s && exec bash %s &' % (self.context.remote_root, script_file_name) )
-        proc.wait()
+        if not isinstance(proc,dict):
+            proc.wait()
+            job_id = int(proc.pid)
+            self.context.write_file(job_id_name, str(job_id))
+            return job_id
+        else:
+            job_id = int(proc["stdout"].readline())
+            self.context.write_file(job_id_name, str(job_id))
+            return job_id
         # proc.kill()
-        job_id = int(proc.pid)
-        self.context.write_file(job_id_name, str(job_id))
-        return job_id
+
```

Comment:
This part is going to be fixed in #82. Please revert it and the rest can merged.